engine/src/voxels/palette.cpp

```diff
@@ -151,53 +156,36 @@ VoxelPalette::Iterator VoxelPalette::end()
     return {mMaterials.end()};
 }
 
-void cubos::core::data::old::serialize(Serializer& serializer, const VoxelPalette& palette, const char* name)
+bool VoxelPalette::loadFrom(Stream& stream)
 {
-    // Count non-empty materials.
-    std::size_t count = 0;
-    for (const auto& material : palette.mMaterials)
-    {
-        if (material.similarity(VoxelMaterial::Empty) < 1.0F)
-        {
-            count++;
-        }
-    }
-
-    serializer.beginDictionary(count, name);
-    for (uint16_t i = 0; i < static_cast<uint16_t>(palette.mMaterials.size()); i++)
+    while (!stream.eof())
```

Comment:
IMO, it's best to store an integer at the start of the file which indicates how many materials the palette will have, instead of expecting the whole stream to have just this palette.
In the future someone using this method may want to store multiple palettes in a single stream, and it should be possible.